solution/index.js

```diff
@@ -0,0 +1,421 @@
+if(!JSON.parse(localStorage.getItem("tasks"))){
+    const tasks = {
+    "todo": [],
+    "in-progress": [],
+    "done": []
+    }
+    localStorage.setItem("tasks", JSON.stringify(tasks));
+}else{
+    const tasks = JSON.parse(localStorage.getItem("tasks"));
+    updateDOMfromLocalStorage()
+}
+
+function createTask(task){
+    const deleteBtnEl = createElement("button",["✖"],["delete-btn"],{},{"click": deleteTask});
+    const additionalSettingsBtnEl = createElement("button",["🔧"],["additional-settings-btn"],{},{"click": openAdditionalSettings});
+    const buttons = createElement("p",[additionalSettingsBtnEl,deleteBtnEl],["task-btns"])
+    const taskP = createElement("p",[task],["task-p"],{contenteditable:"false"})
+    const taskEl = createElement("li",[taskP,buttons],["task","draggable"],{draggable:"true"},{"mouseover": moveTask,"mouseout":stopMove})
+    return taskEl;
+}
+
+function addTask({target}){
+    if(target.className==="add-button"){
+        const task = target.previousSibling.previousSibling.value;  //try to find better js path
+        const listId = target.parentElement.parentElement.id;   //try to find better js path
+        if(task!==""){
+            //create DOM element
+            const taskEl = createTask(task);
+            target.parentElement.after(taskEl);
+            addDragAndDropEventListeners();
+            //Update local storage
+            const tasksObj = JSON.parse(localStorage.getItem("tasks"));
+            tasksObj[listIdToObjKey(listId)].unshift(task);
+            localStorage.setItem("tasks", JSON.stringify(tasksObj));
+            //nullify add input
+            target.previousSibling.previousSibling.value="";
+        }else{
+            alert("Empty Task")
+        }
+        target.blur();
+
+    }
+}
+
+function editTask({target}){
+    if(target.classList.contains("task")){
+        const targetP = target.firstChild;
+        targetP.setAttribute("contenteditable",true);
+        targetP.click();
+        document.querySelector("body").removeEventListener("dblclick",editTask);
+        const task = targetP.innerText; 
+        targetP.addEventListener("blur",(event)=>{updateEditToLocalStorage(event,task);
+        targetP.setAttribute("contenteditable",false)} )
+    }
+    if(target.classList.contains("task-p")){
+        target.setAttribute("contenteditable",true);
+        target.click();
+        document.querySelector("body").removeEventListener("dblclick",editTask);
+        const task = target.innerText; 
+        target.addEventListener("blur",(event)=>{updateEditToLocalStorage(event,task);
+        target.setAttribute("contenteditable",false)} )
+    }
+}
+
+let moveTaskEl;
```

Comment:
Global variables always cause trouble eventually. There are better ways to access the hovered element from the keyboard event handler.